src/components/Agreements/ViewAgreement/ViewAgreement.js

```diff
@@ -135,6 +144,7 @@ class ViewAgreement extends React.Component {
 
     if ((prevAgreement.id !== agreement.id) || (difference(users, prevUsers).length)) {
       this.fetchUsers();
+      this.fetchInvoices();
```

Comment:
Can we move this into it's own condition since it doesn't appear to have anything to do with contacts?

Correct me if I'm wrong, but I'm thinking we want to refetch when either:
- a new agreement has been selected (`id`s are different)
- the number of `financesAgreementLines` is different

Reply:
Thanks for catching this wrong check; changed.